homeassistant/components/tplink/light.py

```diff
@@ -48,29 +48,28 @@
 LIGHT_STATE_ERROR_MSG = "err_msg"
 
 LIGHT_SYSINFO_MAC = "mac"
+LIGHT_SYSINFO_HOST = "host"
 LIGHT_SYSINFO_ALIAS = "alias"
 LIGHT_SYSINFO_MODEL = "model"
 LIGHT_SYSINFO_IS_DIMMABLE = "is_dimmable"
 LIGHT_SYSINFO_IS_VARIABLE_COLOR_TEMP = "is_variable_color_temp"
 LIGHT_SYSINFO_IS_COLOR = "is_color"
 
+MAX_ATTEMPTS = 20
+SLEEP_TIME = 2
+
 
 async def async_setup_entry(hass: HomeAssistantType, config_entry, async_add_entities):
     """Set up switches."""
-    await async_add_entities_retry(
-        hass, async_add_entities, hass.data[TPLINK_DOMAIN][CONF_LIGHT], add_entity
-    )
-    return True
-
+    devices = hass.data[TPLINK_DOMAIN][CONF_LIGHT]
+    entities = []
 
-def add_entity(device: SmartBulb, async_add_entities):
-    """Check if device is online and add the entity."""
-    # Attempt to get the sysinfo. If it fails, it will raise an
-    # exception that is caught by async_add_entities_retry which
-    # will try again later.
-    device.get_sysinfo()
+    for device in devices:
+        await hass.async_add_executor_job(device.get_sysinfo)
+        entities.append(TPLinkSmartBulb(device))
 
-    async_add_entities([TPLinkSmartBulb(device)], update_before_add=True)
+    async_add_entities(entities, update_before_add=True)
+    return True
```

Comment:
Nothing is checking this return value. Please remove the statement.

Reply:
Removing both, this was the part that I was referring to I was confused why some integrations have it.